### PR TITLE
[connectivity_plus] discard unused warnings for `switch reachability.connection ?? .unavailable` in macOS

### PR DESCRIPTION
**Pull request by piedcipher**
## Description

- Discard unused warnings for `switch reachability.connection ?? .unavailable` in macOS by removing `?? .unavailable` as left side of nil coalescing operator '??' has non-optional type 'Reachability.Connection', so the right side is never used.

macOS Implementation does have `??`

Screenshot at 2022-10-03 23-04-05

---

iOS Implementation doesn't have `??`

Screenshot at 2022-10-03 23-03-58

## Related Issues

Fixes: #1078

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by piedcipher** [comment]
iOS - https://github.com/fluttercommunity/plus_plugins/blob/main/packages/connectivity_plus/connectivity_plus/ios/Classes/ReachabilityConnectivityProvider.swift

https://github.com/fluttercommunity/plus_plugins/blob/bccbab9fb663df5cf68af432c60f3357a2111217/packages/connectivity_plus/connectivity_plus/ios/Classes/ReachabilityConnectivityProvider.swift#L8

---

macOS - https://github.com/fluttercommunity/plus_plugins/blob/main/packages/connectivity_plus/connectivity_plus_macos/macos/Classes/ReachabilityConnectivityProvider.swift

https://github.com/fluttercommunity/plus_plugins/blob/bccbab9fb663df5cf68af432c60f3357a2111217/packages/connectivity_plus/connectivity_plus_macos/macos/Classes/ReachabilityConnectivityProvider.swift#L8

---

Content of Both files are same. Except Line 8.

File: packages/connectivity_plus/connectivity_plus/pubspec.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: connectivity_plus
 description: Flutter plugin for discovering the state of the network (WiFi & mobile/cellular) connectivity on Android and iOS.
-version: 2.3.8
+version: 2.3.9
```
Comment (miquelbeltran) [inline packages/connectivity_plus/connectivity_plus/pubspec.yaml:3]: no need to bump the version of the main package when updating the macos one, can you revert this and remove the changelog entry?

Comment (piedcipher) [reply packages/connectivity_plus/connectivity_plus/pubspec.yaml:3]: Sure, my bad. I got confused by 2nd paragraph from contributor.md.

> If updating a platform package (e.g. device_info_plus_macos), please remember to update version and changelog of the main package (e.g. device_info_plus).

Comment (piedcipher) [reply packages/connectivity_plus/connectivity_plus/pubspec.yaml]: Done. Removed version bump from main package and removed changelog entry from main package.

Comment (miquelbeltran) [reply packages/connectivity_plus/connectivity_plus/pubspec.yaml]: yeah, the federated plugin architecture is very confusing, and I'd love to simplify it someday. Thanks for taking care of this!

**Review by miquelbeltran — APPROVED** [review]

**Pull request merged**